### PR TITLE
(886a) Refactor api client and service use for person in prison related information

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

Discussed making some changes prior to creating the offence history page.  Agreed to have a client per external api.

## Changes in this PR

- Renamed `SentenceInformationClient` to `PrisonApiClient`
- Deleted `SentenceInformationService` and moved the sub method into `PersonService` as this information is related to a Person in prison.
- Rename `PrisonClient` to `PrisonRegisterApiClient`
- Rename `PrisonerClient` to `PrisonerSearchClient`
- Removed `CaseloadClient` and moved the only method to `PrisonApiClient`
- Also renamed the mockApi integration test files to be more aligned with the name of the api.

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -7,10 +7,10 @@ import caseloads from './integration_tests/mockApis/caseloads'
 import courses from './integration_tests/mockApis/courses'
 import courseParticipations from './integration_tests/mockApis/courseParticipations'
 import manageUsers from './integration_tests/mockApis/manageUsers'
-import prisons from './integration_tests/mockApis/prisons'
-import prisoners from './integration_tests/mockApis/prisoners'
+import prison from './integration_tests/mockApis/prison'
+import prisonRegister from './integration_tests/mockApis/prisonRegister'
```
Comment (yndajas) [inline cypress.config.ts:11]: These two should probably have `Api` at the end for consistency with the rest of the codebase? I think we should separately remove the "API" from the Prison Register to use its proper name (Prison API, Prison Register, and Prisoner Search appear to be the proper names?)

**Review by yndajas — COMMENTED** [review]
Looks good. Is there a reason not to do the following in this PR as well?
- CaseloadClient -> PrisonApiClient
- PrisonClient -> PrisonRegisterApiClient
- PrisonerClient -> PrisonerSearchClient
- and maybe moving `CourseClient` and `ReferralClient` into an `accreditedProgrammesApi` folder

I also added a comment about moving from Prison Register API to Prison Register which could be dealt with here or in a separate PR later

File: server/services/personService.test.ts [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
 
 jest.mock('../data/prisonerClient')
 jest.mock('../data/hmppsAuthClient')
+jest.mock('../data/prisonApiClient')
 jest.mock('../utils/personUtils')
 
 const redisClient = createMock<RedisClient>({})
 const tokenStore = new TokenStore(redisClient) as jest.Mocked<TokenStore>
-const systemToken = 'some system token'
+const systemToken = 'some system systemToken'
```
[the remark this answers is not in the document]
Comment (jsrobertson) [reply server/services/personService.test.ts:23]: Nope :D

File: server/data/prisonApiClient.ts [hunk 3]
```diff
@@ -17,4 +17,10 @@ export default class PrisonApiClient {
       path: prisonApiPaths.sentenceAndOffenceDetails({ bookingId }),
     })) as SentenceAndOffenceDetails
   }
+
+  async getCurrentUserCaseloads(): Promise<Array<Caseload>> {
```
Comment (yndajas) [inline server/data/prisonApiClient.ts:21]: `findCaseloadsByCurrentUser`? Most of our client methods use `find`

**Review by yndajas — APPROVED** [review]
One naming suggestion, but otherwise looks good to me!

**Pull request merged**